### ISSUE TITLE
Extract/Move columns modal breaks entirely when all columns are selected

### ISSUE THREAD

**Issue by pavish** [mathesar-foundation/mathesar#3339]
## Description
* Case 1:
  * Select a column in a table, and click 'Extract Columns into a New Table' or 'Move Columns to Linked Table' option in the Table inspector -> Column tab.
  * Select all the columns in the modal. Observe that the modal automatically closes.
* Case 2:
  * Select all the columns in a table for a row, and click the 'Extract Columns into a New Table' or 'Move Columns to Linked Table' option.
  * Try changing the columns in the modal. The modal is unresponsive and breaks the entire frontend.

## Expected behavior
* The  'Extract Columns into a New Table' and 'Move Columns to Linked Table' UI should not break, and should show a valid error if all columns are selected.

## Video recording

https://github.com/mathesar-foundation/mathesar/assets/11850603/dbc6cf0a-693d-474b-874c-83afcc7165b0

**Comment by kanugurajesh** [comment]
Hi I want to work on this issue. Is this issue still open

**Comment by seancolsen** [comment]
Yes @kanugurajesh this is still open. Feel free to submit a PR. (Note that as a policy we don't assign issues to new contributors.)

**Comment by GopalVerma1303** [comment]
Hey @kanugurajesh! Are u working on this issue? If not, I m interested in giving it a try @seancolsen.

**Comment by seancolsen** [comment]
Feel free to submit a PR, @GopalVerma1303

**Comment by commonspider** [comment]
I'm trying to replicate both case 1 and 2, but the bug doesn't appear.
Is this issue already solved?

**Comment by seancolsen** [comment]
Thanks for following up, @commonspider!

I've tried reproducing this using the exact data and steps shown in the video, but I'm also unable to reproduce it.

Here a partial theory which could explain our inability to reproduce it:

- It's possible this bug was related to the cell-selection logic, since the cell selection is automatically updated while the user interacts with this form.
- I refactored that logic in #3037, which was merged after this ticket was opened. That PR fixed a lot of miscellaneous bugs related to cell selection. Perhaps it fixed this too!

I'm closing, since this seems both low-priority and possibly fixed.
